lib/codeowners/checker.rb

```diff
@@ -34,11 +34,8 @@ def added_files
       changes_to_analyze.select { |_k, v| v == 'A' }.keys
     end
 
-    def check!
-      @results ||= {
-        missing_ref: missing_reference,
-        useless_pattern: useless_pattern
-      }
+    def fix!
```

Comment:
This name is hard to understand what it does. Can you rename the method or add some user documentation?

I'd vote to rename to `start_interactive_mode` or `catch_user_quit`. Or even wrap the `if options[:interactive]` and rename to `setup_interactive_options`.

I still don't get why we need to return the `results` apart from testing it.

Reply:
using one `results` allows to keep it consistent as to which methods need to be executed, the return value is side effect here not the expected behavior

with example results as configurable we would have the selection process done in one place, this way whatever we run `fix!` or `consistent?` we know they are related
```ruby
    def results(missing_references: false, useless_patterns: false)
      all = !missing_references && !useless_patterns
      @results = {}.tap do |hash|
        hash[missing_ref]     = missing_reference if all || missing_references
        hash[useless_pattern] = useless_pattern   if all || useless_patterns
      end
    end
```